### ISSUE TITLE
Tracer panic for ICMP extension

### ISSUE THREAD

**Issue by fujiapple852** [fujiapple852/trippy#1287]
Tracer panic:

```
thread 'tracer-23144' panicked at crates/trippy-packet/src/icmp_extension.rs:426:33:
range end index 42973 out of range for slice of length 416
```

Attempting to reproduce and capture the offending packet.

**Comment by fujiapple852** [comment]
Unable to reproduce whilst capturing packets.  Instead I created a synthetic test case based on the error message and fixed based on that.  Provisional fix in #1294.

**Comment by fujiapple852** [comment]
Still unable to reproduce despite exhaustive attempts.  

The change in https://github.com/fujiapple852/trippy/pull/1294 fixes _an_ issue and even if it does not fix _the_ issue (it almost certainly does) it is still a valid fix and will be merged.
